### PR TITLE
Full refactor

### PR DESCRIPTION
**Pull request by cef62**
Completely rewritten implementation of the library to benefit of the new [`react-styleguidist`](https://github.com/styleguidist/react-styleguidist/releases/tag/v4.5.0) API to configure the undelying `express` server.

Also the setup is reduced to a single action: just process the project `styleguide.config.js` module to `snapguidist`:  

```javascript
const snapguidist = require('snapguidist')

module.exports = snapguidist({ [...] })
```

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
```
Comment (MicheleBertoli) [inline .babelrc]: We don't need this file.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,6 @@
 dist
```
Comment (MicheleBertoli) [inline .gitignore]: This should be:
```
.opt-out
.opt-in
.snapguidist
node_modules
npm-debug.log
```

File: example/package.json [hunk 3]
```diff
@@ -3,14 +3,15 @@
...
   },
   "devDependencies": {
-    "babel-core": "^6.20.0",
-    "babel-loader": "^6.2.10",
-    "react": "^15.4.1",
-    "react-dom": "^15.4.1",
-    "react-styleguidist": "^4.4.1",
-    "webpack": "^1.14.0"
+    "babel-core": "6.20.0",
+    "babel-loader": "6.2.10",
```
Comment (MicheleBertoli) [inline example/package.json]: We don't need `babel-core` and `babel-loader`.

File: example/styleguide.config.dev.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+const path = require('path')
+const snapguidist = require(path.join(__dirname, '..', 'src', 'index'))
+
+
```
Comment (MicheleBertoli) [inline example/styleguide.config.dev.js]: Unnecessary new line.

File: package.json [hunk 5]
```diff
@@ -1,61 +1,62 @@
...
-    "react-addons-test-utils": "^15.4.1",
-    "react-dom": "^15.4.1",
-    "react-styleguidist": "^4.4.1",
-    "rimraf": "^2.5.4",
-    "supertest": "^2.0.1",
-    "supertest-as-promised": "^4.0.2"
+    "babel-jest": "18.0.0",
+    "babel-preset-es2015": "6.18.0",
+    "babel-preset-react": "6.16.0",
+    "babel-preset-stage-0": "6.16.0",
```
Comment (MicheleBertoli) [inline package.json]: We don't need `babel-preset-stage-0`.

File: package.json [hunk 6]
```diff
@@ -1,61 +1,62 @@
...
+    "enzyme": "2.7.0",
+    "enzyme-to-json": "1.4.5",
+    "eslint": "3.12.2",
+    "eslint-config-airbnb": "13.0.0",
+    "eslint-plugin-import": "2.2.0",
+    "eslint-plugin-jsx-a11y": "2.2.3",
+    "eslint-plugin-react": "6.8.0",
+    "express": "4.14.0",
```
Comment (MicheleBertoli) [inline package.json]: We don't need `express`.

File: package.json [hunk 7]
```diff
@@ -1,61 +1,62 @@
...
-    "react-test-renderer": "^15.4.1"
+    "body-parser": "1.15.2",
+    "classnames": "2.2.5",
+    "cors": "2.8.1",
+    "jest-snapshot": "18.1.0",
+    "react-test-renderer": "15.4.1"
   },
   "peerDependencies": {
-    "react-styleguidist": "^4.4.1"
+    "react": ">15",
```
Comment (MicheleBertoli) [inline package.json]: This should be:
```
"react": ">=15",
```

File: src/core/updateWebpackConfig.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+const PLAYGROUND_PREVIEW = 'rsg-components/Preview'
+
+function updateWebpackConfig(webpackConfig, env, serverInfo) {
+  webpackConfig.module.loaders.push(
+    {
+      test: /\.jsx?$/,
+      include: srcFolder,
+      loader: 'babel',
+      query: {
+        presets: [
```
Comment (MicheleBertoli) [inline src/core/updateWebpackConfig.js]: This should be:
```
presets: [
  'es2015',
  'react',
],
```

File: src/core/updateWebpackConfig.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+      },
+    },
+    {
+      test: /\.css$/,
+      include: srcFolder,
+      loaders: ['style', 'css'],
+    }
+  )
+
+  // TODO: should be avoided to add style.css as last item in `entry`?
```
Comment (MicheleBertoli) [inline src/core/updateWebpackConfig.js]:
```
// TODO: please remove TODOS :)
```

File: example/styleguide.config.dev.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+module.exports = snapguidist({
+  title: 'Snapguidist Development Example',
```
Comment (MicheleBertoli) [inline example/styleguide.config.dev.js]: I'd try to keep the name consistent, what do you think about:
> [dev] Snapguidist Styleguide

File: src/components/PlaygroundRenderer.js [hunk 11]
```diff
@@ -16,8 +16,8 @@ class SnapguidistPlaygroundRenderer extends Component {
 }
 
 SnapguidistPlaygroundRenderer.propTypes = {
-  code: PropTypes.string.isRequired,
-  evalInContext: PropTypes.func.isRequired,
+  // code: PropTypes.string.isRequired,
```
Comment (MicheleBertoli) [inline src/components/PlaygroundRenderer.js]: Not sure about this comments, should we remove the `propTypes`?

File: example/styleguide.config.dev.js (hunk 10 above)
Comment (cef62) [inline example/styleguide.config.dev.js]: I like that ;)

File: .babelrc (hunk 1 above)
Comment (cef62) [inline .babelrc]: I agree the source doesn't need it but we have to keep it to run our tests ;)

Comment (MicheleBertoli) [inline .babelrc]: Aha, we need it for the tests. Thanks @cef62 :)
Just remove `stage-0`.

**Pull request merged**